frmts/raw/nsidcbindataset.cpp

```diff
@@ -0,0 +1,496 @@
+
+class NSIDCbinDataset final: public GDALPamDataset
+{
+  friend class NSIDCbinRasterBand;
+  struct NSIDCbinHeader{
+
+    // page 7, User Guide https://nsidc.org/data/nsidc-0051
+    // 1.3.2 File Contents
+    // The file format consists of a 300-byte descriptive header followed by a two-dimensional array of
+    // one-byte values containing the data. The file header is composed of:
+    // - a 21-element array of 6-byte character strings that contain information such as polar
+    // stereographic grid characteristics
+    // - a 24-byte character string containing the file name
+    // - a 80-character string containing an optional image title
+    // - a 70-byte character string containing ancillary information such as data origin, data set
+    // creation date, etc.
+    // For compatibility with ANSI C, IDL, and other languages, character strings are terminated with a
+    // NULL byte.
+    // Example file:
+    // ftp://sidads.colorado.edu/pub/DATASETS/nsidc0051_gsfc_nasateam_seaice/final-gsfc/south/daily/2010/nt_20100918_f17_v1.1_s.bin
+
+    char missing_int[6] = {0};      // "00255"
+    char columns[6] = {0};          // "  316"
+    char rows[6] = {0};             // "  332"
+    char internal1[6] = {0};        // "1.799"
+    char latitude[6] = {0};         // "-51.3"
+    char greenwich[6] = {0};        // "270.0"
+    char internal2[6] = {0};        // "558.4"
+    char jpole[6] = {0};            // "158.0"
+    char ipole[6] = {0};            // "174.0"
+    char instrument[6] = {0};       // "SSMIS"
+    char data_descriptors[6] = {0};       // "17 cn"
+    char julian_start[6] = {0};     // "  261"
+    char hour_start[6] = {0};       // "-9999"
+    char minute_start[6] = {0};     // "-9999"
+    char julian_end[6] = {0};       // "  261"
+    char hour_end[6] = {0};         // "-9999"
+    char minute_end[6] = {0};       // "-9999"
+    char year[6] = {0};             // " 2010"
+    char julian[6] = {0};           // "  261"
+    char channel[6] = {0};          // "  000"
+    char scaling[6] = {0};          // "00250"
+
+    // 121-126 Integer scaling factor
+    // 127-150 24-character file name (without file-name extension)
+    // 151-230 80-character image title
+    // 231-300 70-character data information (creation date, data source, etc.)
+    char filename[24] = {0};        // "  nt_20100918_f17_v1.1_s"
+    char imagetitle[80] = {0};  // "ANTARCTIC  SMMR  TOTAL ICE CONCENTRATION       NIMBUSN07     DAY 299 10/26/1978"
+    char data_information[70] = {0};    // "ANTARCTIC  SMMR ONSSMIGRID CON Coast253Pole251Land254      06/27/1996"
+
+    };
+
+    VSILFILE    *fp = nullptr;
+    CPLString    osSRS{};
+    NSIDCbinHeader  sHeader{};
+    GDALDataType eRasterDataType;
+    double      adfGeoTransform[6];
+    CPL_DISALLOW_COPY_ASSIGN(NSIDCbinDataset)
+
+  public:
+    NSIDCbinDataset();
+    ~NSIDCbinDataset() override;
+    CPLErr GetGeoTransform( double * ) override;
+    OGRSpatialReference oSRS;
```

Comment:
oSRS should be moved to the private section, and initialized with {}
By the way, in new code, we tend to prefix member variables with m_ to avoid potential confusion with local variables